### ISSUE TITLE
Peer Review

### ISSUE THREAD

**Issue by aviatoali** [csoulakian/cs330s16p1#2]
Yeah, I really gotta agree with Tyler on the commenting, it's very thorough and helpful.

Don't really know what to add since the code looks more sophisticated and smoother than anything I thought of. One thing to think about that gave me pause though is the benefits of parsing the string directly, rather than parsing a String array of words after a split. I considered a split array in the beginning as well, but if you ever want to add the functionality of determining if the post is a reply or not, which might be an important piece of information, it might take a bit more work post string-split.

 For tweets a mention @ at the very beginning of the post categorizes the tweet as a reply to mentioned user. I imagine if the professor asks us to expand the project later to store data for multiple, possibly related tweets, methods to recognize that a tweet is a reply, who it is a reply to and connect the tweet to the replied user will be important. In that expanded functionality parsing the entirety of the message string allows use of startsWith() methods to recognize a reply immediately. Like, your sorter that matches using the regMentions String would just need an extra if to check if matched mention has index 0 in message string. 

Otherwise, if parsing by split words you'd have to either include a separate method for checking if first mention is reply and storing username of replied-to user. 

TL;DR: Can't really add much since your code and documentation look great as is. Consider switching to a full string parse rather than split word parsing. Allows more discerning sorting, imo. Doubt it makes too much of time difference anyway, if any.

**Comment by csoulakian** [comment]
Thanks for the feedback! You make a great point and I didn't consider the position of the mentions. Depending on where the project leads, I might end up needing to parse the whole message, but I'll leave it as is for now.
